### PR TITLE
[CCTRI-2906] Add new endpoints into functional testing

### PR DESCRIPTION
**Pull request by YevhenLysen**
Added new tests for the new endpoints

### CODE REVIEW

File: tests/functional/tests/test_ctia_api.py [hunk 1]
```diff
@@ -117,985 +122,4366 @@ def test_python_module_ctia_positive_actor(module_headers, module_tool_client):
...
-
-def test_python_module_ctia_positive_bulk(module_headers, module_tool_client):
-    """Perform testing for bulk functionality of custom threat intelligence
-    python module
+    delayed_return(actor.search.delete(params={'id': entity_id,
+                                'REALLY_DELETE_ALL_THESE_ENTITIES': 'true'}))
+    # Repeat GET request and validate that entity was deleted
+    assert actor.search.get(params={'id': entity_id}) == []
+    # Count entities after entity deleted
+    get_actor_search_count2 = actor.search.count()
```
Comment (TarasMaliuchenko) [inline tests/functional/tests/test_ctia_api.py:194]: Think of a more appropriate name for this variable. From the name of the variable it should be clear what value it stores.

Comment (YevhenLysen) [reply tests/functional/tests/test_ctia_api.py]: Thank you. Names of variables were changed.

File: tests/functional/tests/test_ctia_api.py [hunk 2]
```diff
@@ -117,985 +122,4366 @@ def test_python_module_ctia_positive_actor(module_headers, module_tool_client):
         actor.get(entity_id)
 
 
-def test_python_module_ctia_positive_attack_pattern(
-        module_headers, module_tool_client):
-    """Perform testing for attack pattern entity of custom threat intelligence
-    python module
+def test_python_module_ctia_positive_actor_search(module_headers,
```
[the remark this answers is not in the document]
Comment (YevhenLysen) [reply tests/functional/tests/test_ctia_api.py:125]: Thank you. Unused parameters were removed from tests.

File: tests/functional/tests/test_ctia_api.py [hunk 3]
```diff
@@ -117,985 +122,4366 @@ def test_python_module_ctia_positive_actor(module_headers, module_tool_client):
...
-    delayed_return(attack_pattern.delete(entity_id))
-    with pytest.raises(HTTPError):
-        attack_pattern.get(entity_id)
-
-
-def test_python_module_ctia_positive_bulk(module_headers, module_tool_client):
-    """Perform testing for bulk functionality of custom threat intelligence
-    python module
+    delayed_return(actor.search.delete(params={'id': entity_id,
+                                'REALLY_DELETE_ALL_THESE_ENTITIES': 'true'}))
```
[the remark this answers is not in the document]
Comment (YevhenLysen) [reply tests/functional/tests/test_ctia_api.py]: Thank you. Issue related to flake8 were fixed.

File: tests/functional/tests/test_ctia_api.py [hunk 4]
```diff
@@ -117,985 +122,4366 @@ def test_python_module_ctia_positive_actor(module_headers, module_tool_client):
...
+    # Repeat GET request and validate that entity was deleted
+    assert actor.search.get(params={'id': entity_id}) == []
+    # Count entities after entity deleted
+    get_actor_search_count2 = actor.search.count()
+    # Compare results of get_actor_search_count and get_actor_search_count2
+    assert get_actor_search_count != get_actor_search_count2
+
+
+def test_python_module_ctia_positive_actor_metric(module_headers,
```
[the remark this answers is not in the document]
Comment (YevhenLysen) [reply tests/functional/tests/test_ctia_api.py]: Thank you. Unused 'module_headers' parameters were removed from tests.

**Review by MaxMhonc — APPROVED** [review]

**Review by TarasMaliuchenko — APPROVED** [review]

**Pull request merged**